ministryofjustice/correspondence_tool_staff
# CT-2228 Basic word export working

## Description
User needs to be able to download the generated letter as a Word document. Added an alternate rendering format for :docx, a link hitting same, and amended the show action to produce the document. Some basic tests and a little refactoring of the Letter class so it now wraps letter template and Case. 
 
## Self-review checklist

 
### Screenshots

![image](https://user-images.githubusercontent.com/1161161/68189079-899c0880-ffa2-11e9-82cb-01292b4fbfd5.png)

![image](https://user-images.githubusercontent.com/1161161/68189107-97ea2480-ffa2-11e9-85b6-df300330caff.png)


 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2228
 
### Deployment
We need to run the seeder again, as some of the balance of stuff in the database versus the Word template/view file has changed a bit. `rake db:seed:dev:letter_templates` locally if you want to try it out, `rake db:seed:letter_templates` required on the server after deployment. 
 
### Manual testing instructions
Case in Waiting for data - press the "Send acknowledgement letter" button. Choose the template, view the letter, click "Save as Word" to download a file and check the replaced fields look sensible
